pkg/container/ring.go

```diff
@@ -16,22 +16,22 @@
 // readers.
 //
 // To iterate over a PayloadRing (where t is the PayloadRing):
-// 	for lastWritePos := t.LastWriteParallel(); ; lastWritePos-- {
-// 	    value, stop := s.Read(lastWriteIdx)
+//	for lastWritePos := t.LastWriteParallel(); ; lastWritePos-- {
+//	    value, stop := s.Read(lastWriteIdx)
 //      if stop {
 //          break
 //      }
 //      // do something with value
-// 	}
+//	}
 //
 // Write is *not* being executed in parallel you can use
-// 	for lastWritePos := t.LastWrite(); ; lastWritePos-- {
-// 	    value, stop := s.Read(lastWriteIdx)
+//	for lastWritePos := t.LastWrite(); ; lastWritePos-- {
+//	    value, stop := s.Read(lastWriteIdx)
 //      if stop {
 //          break
 //      }
 //      // do something with value
-// 	}
+//	}
```

Comment:
Can probably update docs to use the reader. In fact, we may want to hide the ring to be private and only expose the reader